41 Tanslate dataset math problems

jansowa opened the pull request:
Scripts for #41 
I will send the dataset soon, I will just make some manual corrections.

jansowa committed:
First version of scripts to generate Polish orca math translation

jansowa committed:
Polish orca math - add samples and fix silly errors

jansowa committed:
Polish orca math - add documentation

IgorTest19 commented on instructions_scripts/orca_math_create_english_docx.py:
```diff
@@ -0,0 +1,78 @@
+from datasets import load_dataset
```
It would be nice to include a docstring of the script describing what it is reponsible for   and move there comments which are below imports.

pawelkiszczak replied:
I believe it can be added in future version as long as the current one operates properly.

jansowa replied:
I've done it. Additionaly, I've added more detailed documentation.

pawelkiszczak replied:
Changes made, guessing we can merge now @IgorTest19

IgorTest19 commented on instructions_scripts/orca_math_create_json_from_docx.py:
```diff
@@ -0,0 +1,117 @@
+import re
```
It would be nice to include a docstring of the script describing what it is reponsible for and move there comments which are below imports.

pawelkiszczak replied:
As above - as long as it's working properly, docstring can be updated in future commits.

jansowa replied:
As above - finished with more detailed documentation.

IgorTest19 commented on instructions_scripts/orca_math_create_english_docx.py:
```diff
@@ -0,0 +1,78 @@
+# This part only creates small .docx files from the dataset, which can then be conveniently translated using external services.
+# Later create a dataset from the translated documents using orca_math_create_json_from_docx.py
+
+dataset = load_dataset("microsoft/orca-math-word-problems-200k")
```
If downloading of the dataset is not automated/implemented in the script, it would be good to include information about the need/requirement of downloading it manually from the provided link.

pawelkiszczak replied:
Seems like downloading is handled by `load_dataset` method and dataset is being downloaded from Hugging Face.

jansowa replied:
@IgorTest19 , "microsoft/orca-math-word-problems-200k" is not a local directory. As @pawelkiszczak said - it's handled by huggingface dataset.load_dataset. The method downloads it from https://huggingface.co/datasets/microsoft/orca-math-word-problems-200k.

I can download it directly from https://huggingface.co/datasets/microsoft/orca-math-word-problems-200k/blob/refs%2Fconvert%2Fparquet/default/train/0000.parquet , but I'm not sure that it's a good idea.

IgorTest19 replied:
I've missed that it is being executed by the third party library, not our own implementation. That's fine by me then :)

IgorTest19 commented on instructions_scripts/orca_math_create_json_from_docx.py:
```diff
@@ -0,0 +1,117 @@
+    output_file.write('\n]')
+
+print("Combined file saved into " + combined_file_path)
+print("You can now delete all other files created during translation.")
```
We may consider creating a standalone directory for instructions scripts that require manual work, such as translation, since this is not fully automated (in this case). We can create a directory `instructions_manual`/`instructions_scripts_manual`:
```
speakleash-instruct-creator/
├── instructions_scripts/
├── instructions_manual/
│ ├── orca_math_create_english_docx.py
│ └── orca_math_create_json_from_docx.py
├── instructions_samples/
│ └── polish_orca_math_sample.json
├── requirements.txt
⋮
```

pawelkiszczak replied:
Seems like a good idea, we can go in this direction.

jansowa replied:
Yep, that's a good idea. Done :)

pawelkiszczak reviewed (commented):
Comments as can be seen, please let me know future steps and shall those be resolved now or in future commits.

jansowa committed:
Polish orca math:
- move scripts to the new directory (as they require additional manual tasks)
- more detailed documentation, moved to the top of files

jansowa committed:
Merge branch 'speakleash:main' into 41-translate-dataset-math-problems

pawelkiszczak commented:
Everything seems resolved, go for merge please @IgorTest19
